src/lava/lib/dl/netx/hdf5.py

```diff
@@ -0,0 +1,458 @@
+from lava.lib.dl.netx.utils import optimize_weight_bits
+from lava.lib.dl.netx.blocks.process import Input, Dense, Conv
+from lava.lib.dl.netx.blocks.models import AbstractPyBlockModel
+
+
+class Network(AbstractProcess):
+    """Generates a Lava process for the network described in hdf5 config.
+
+    Parameters
+    ----------
+    net_config : str
+        name of the hdf5 config file.
+    num_layers : int, optional
+        number of blocks to generate. An integer valuew will only generate the
+        first ``num_layers`` blocks in the description. The actual number of
+        generated layers may be less than ``num_layers``. If it is None, all
+        the layers are generated. Defaults to None.
+    has_graded_input : bool, optional
+        flag indicating the input spike type of input layer. Defaults to False.
+    """
+    def __init__(
+        self,
+        **kwargs: Union[int, Tuple[int, ...]]
+    ) -> None:
+        super().__init__(**kwargs)
+        self.filename = kwargs.pop('net_config')
+        self.net_config = NetDict(self.filename)
+
+        self.num_layers = kwargs.pop('num_layers', None)
+        self.has_graded_input = kwargs.pop('has_graded_input', False)
+
+        self.net_str = ''
+        self.layers = self._create()
+
+        self.in_layer = self.layers[0]
+        self.out_layer = self.layers[-1]
+
+        self.inp = InPort(shape=self.in_layer.inp.shape)
+        self.out = OutPort(shape=self.out_layer.out.shape)
+
+        self.inp.connect(self.in_layer.inp)
+        self.out_layer.out.connect(self.out)
+
+        self.has_graded_input = self.in_layer.has_graded_input
+        self.has_graded_output = self.out_layer.has_graded_output
+
+    def __str__(self) -> str:
+        """Network description string."""
+        return self.net_str
+
+    def __len__(self) -> int:
+        """Number of layers in the network."""
+        return len(self.layers)
+
+    @staticmethod
+    def get_neuron_params(
+        neuron_config: h5py.Group,
+        input: bool = False,
+    ) -> AbstractProcess:
+        """Provides the correct neuron configuration process and parameters
+        from the neuron description in hdf5 config.
+
+        Parameters
+        ----------
+        neuron_config: h5py.Group
+            hdf5 object describing the neuron configuration
+        input: bool
+            flag to indicate if the layer is input. For some cases special
+            processing may be done.
+
+        Returns
+        -------
+        AbstractProcess
+            The Lava process that implements the neuron described.
+        """
+        neuron_type = neuron_config['type']
+        if neuron_type in ['LOIHI', 'CUBA']:
+            neuron_process = LIF
+            neuron_params = {
+                'neuron_proc': neuron_process,
+                'vth': neuron_config['vThMant'],
+                'du': neuron_config['iDecay'] - 1,
+                # weird: it seems the LIF process is written this way!
```

Comment:
Probably should revise this or remove it.

Reply:
Maybe add an issue in Lava if something needs to change with LIF process.

Reply:
I'll remove it. It's got to do with the default value of registers in our hardware. I always find it unintuitive to treat voltage decay and current decay differently!